Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1100 @@
+                                    $frontDoorEp.ResourceGroup = $resourceGroupName
+                                    $frontDoorEp.TypeOfCertificate = $TypeOfCertificate
+                                    $frontDoorEp.isMinTLSVersionSetOnCustomDomain = $false
+                                    
+                                    if($null -ne $frontDoorEp)
+                                    {
+                                        $frontDoorsEndpointWithoutReqMinTLSVersion +=  $frontDoorEp | Select-Object @{N='ResourceGroupName';E={$frontDoorEp.ResourceGroup}},
+                                                                                        @{N='ResourceName';E={$frontDoorEp.FrontDoorName}},
+                                                                                        @{N='DomainName';E={$frontDoorEp.DomainName}},
+                                                                                        @{N='CertificateType';E={$frontDoorEp.TypeOfCertificate}},
+                                                                                        @{N='MinimumTlsVersion';E={$frontDoorEp.MinTLSVersion}},
+                                                                                        @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorEp.isMinTLSVersionSetOnCustomDomain}}
+                                    }
+                                }
+                                else
+                                {
+                                    $frontDoorsEndpointWithtReqMinTLSVersion += $item
+                                }
+                            }
+                            else
+                            {
+                                Write-Host "Skipping this Front End Point..." -ForegroundColor $([Constants]::MessageType.Warning)
+                                Write-Host $([Constants]::SingleDashLine)
+                            }
+                        }
+                }
+            }
+            catch
+            {
+                $frontDoorsSkipped += $frontDoorResource
+                Write-Host "Error fetching Front Door configuration: Resource Name: [$($resourceName)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Encountered error while fetching Front Door configuration")    
+                $logSkippedResources += $logResource
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    
+
+   foreach($item in $frontDoorsEndpointWithoutReqMinTLSVersion)
+   {
+        if ($item.MinimumTlsVersion -eq $requiredMinTLSVersion)
```

Comment:
Will any endpoint will meet this condition as all endpoints in this collection, 'frontDoorsEndpointWithoutReqMinTLSVersion' will definitely not have required version.

Reply:
Fixed

Reply:
Condition $item.MinimumTlsVersion -eq $requiredMinTLSVersion is till there, do we really need this?

Reply:
This condition is used to filter out one domain and write a message on ui with the hostname

Reply:
this could be merged to the above loop but removing this will again need a round of testing.

Reply:
Removed this condition and merged the code.